core/src/mender-artifact.c

```diff
@@ -504,10 +504,14 @@ mender_artifact_release_ctx(mender_artifact_ctx_t *ctx) {
 #ifdef CONFIG_MENDER_FULL_PARSE_ARTIFACT
         mender_utils_key_value_list_free(ctx->artifact_info.provides);
         mender_utils_key_value_list_free(ctx->artifact_info.depends);
-        for (mender_artifact_checksum_t *checksum = ctx->artifact_info.checksums; NULL != checksum; checksum = checksum->next) {
-            free(checksum->filename);
+        mender_artifact_checksum_t *next;
+        for (mender_artifact_checksum_t *checksum = ctx->artifact_info.checksums; NULL != checksum; checksum = next) {
+            FREE_AND_NULL(checksum->filename);
```

Comment:
This can do `free(checksum->filename)`, no point in setting some value to `NULL` when we are destroying it 4 lines below. But not a biggie.